exchange/bitswap/wantmanager.go

```diff
@@ -168,11 +168,17 @@ func (mq *msgQueue) runQueue(ctx context.Context) {
 }
 
 func (pm *WantManager) Connected(p peer.ID) {
-	pm.connect <- p
+	select {
+	case pm.connect <- p:
+	case <-pm.ctx.Done():
+	}
 }
```

Comment:
On Fri, Jun 12, 2015 at 02:49:27PM -0700, Jeromy Johnson wrote:
> these are used to inform bitswap that a peer has connected or
> disconnected.

PeerConnected shows up in the Receiver interface in
exchange/bitswap/network/interface.go too, but I'm still not seeing
the code that checks this during a bitswap *Block call.